App/Sagas/TextileSagas.js

```diff
@@ -239,24 +239,52 @@ export function * shareImage ({thread, hash, caption}) {
 
 export function * photosTask () {
   try {
-    const photos = yield call(queryPhotos)
-    for (const photo of photos.reverse()) {
+    const camera = yield select(TextileSelectors.camera)
+    const processed = camera && camera.processed ? camera.processed : []
+
+    let allPhotos = yield call(getAllPhotos)
+    if (camera === undefined || camera.processed === undefined || processed.length === 0) {
+      const ignoredPhotos = allPhotos.splice(1)
+      for (const uri of ignoredPhotos.map((photo) => photo.uri)) {
+        yield put(TextileActions.imageIgnore(uri))
+      }
+    }
+
+    let allProcessed = processed.reduce((o, item, index) => ({...o, [item]: { index }}), {})
+    const photos = allPhotos.filter((photo) => {
+      if (allProcessed[photo.uri]) {
+        return false
+      }
+      return true
+    })
+
+    for (const orig of photos.reverse()) {
+      const photo = yield call(scalePhoto, orig)
       const multipartData = yield call(IPFS.addImageAtPath, photo.path, photo.thumbPath, 'default')
-      yield call(RNFS.unlink, photo.path)
-      yield call(RNFS.unlink, photo.thumbPath)
-      yield put(TextileActions.imageAdded('default', multipartData.boundary, multipartData.payloadPath))
-      yield put(IpfsNodeActions.getPhotoHashesRequest('default'))
-      yield call(
-        Upload.startUpload,
-        {
-          customUploadId: multipartData.boundary,
-          path: multipartData.payloadPath,
-          url: 'https://ipfs.textile.io/api/v0/add?wrap-with-directory=true',
-          method: 'POST',
-          type: 'multipart',
-          field: multipartData.boundary
+      try { // single photo try/catch
+        yield call(RNFS.unlink, photo.path)
+        yield call(RNFS.unlink, photo.thumbPath)
+        yield put(TextileActions.imageAdded(photo.uri, 'default', multipartData.boundary, multipartData.payloadPath))
+        yield put(IpfsNodeActions.getPhotoHashesRequest('default'))
+        yield call(
+          Upload.startUpload,
+          {
+            customUploadId: multipartData.boundary,
+            path: multipartData.payloadPath,
+            url: 'https://ipfs.textile.io/api/v0/add?wrap-with-directory=true',
+            method: 'POST',
+            type: 'multipart',
+            field: multipartData.boundary
+          }
+        )
+      } catch (error) {
+        try { // no matter, what. try to clean up photos
+          yield call(RNFS.unlink, photo.path)
```

Comment:
These calls to `unlink` seem redundant. Maybe the better structure for the singe photo try/catch would be, first remove the calls to `unlink` inside the `try`. Then the whole thing could look something like:
```
try {
  // scale, add, put imageAdded, startUpload
} catch {
  // put photoProcessingError
} finally {
  // unlink calls
}
```